util/file_util.py

```diff
@@ -811,8 +833,17 @@ def file_open_google_spreadsheet(url: str) -> gspread.spreadsheet.Spreadsheet:
     google spreadsheet object for the given url
   """
     # Get a handle for the whole spreadsheet
-    gs = _file_get_gspread_client().open_by_url(url)
-    return gs
+    while retries >= 0:
```

Comment:
optional: should we use retry module  https://pypi.org/project/retry/?

Reply:
Done